tests/test_hello_world.py

```diff
@@ -73,37 +71,83 @@ def test_fibonacci_using_fixture(first_fibonacci_numbers):
 def please_test_me(string: str) -> str:
     return string + "!!!"
 
+def test_please_test_me():
+    assert please_test_me("testing is great") == "testing is great!!!"
+    assert please_test_me("I love Pycharm") == "I love Pycharm!!!"
+    assert please_test_me("this will work?") == "this will work?!!!"
+
 
 def times_7(number: Union[int, float]):
     return number * 7
 
-
 # TODO make_me_2_functions_one_use_fixture_and_one_use_parametrize
-def test_make_me_2_functions_one_use_fixture_and_one_use_parametrize():
-    assert times_7(2) == 14
-    assert times_7(4) == 28
-    assert times_7(0) == 0
-    assert times_7(-1) == -7
-    # TODO add one interesting case I didn't check
 
+@pytest.mark.parametrize("num", [2,100,-1,-3,0, -5])
+def test_times_7_parametrize(num):
+    assert times_7(num) == 7*num
+    assert times_7(num) / 7 == num
+
+@pytest.fixture
+def rnd_num():
     random_generator = random.Random()
+    return random_generator.randint(-1000, 1000)
+
+def test_times_7_fixture(rnd_num):
+
     for i in range(10):
-        rnd_int = random_generator.randint(-1000, 1000)
         # time_7(rnd_int) is like summing 7 items of rnd_int
-        assert times_7(rnd_int) == sum([rnd_int for i in range(7)])
+        assert times_7(rnd_num) == sum([rnd_num for i in range(7)])
 
         # assert times_7(rnd_int) > rnd_int  # TODO Explain why this assert doest work
-
+        # This assert won't work since rnd_int can be negative thus rnd_int*7 < rnd_int
 
 # TODO Add a function and at least 3 tests
+def dist(x: tuple, y: tuple) -> float:
+    return math.sqrt((x[0] - y[0])**2 + (x[1] - y[1])**2)
+
+@pytest.mark.parametrize("x, y", [((1,1), (0,0)), ((2,5), (0,0)), ((-8,3), (0,0))])
+def test_dist(x, y):
+    assert round(dist(x,y), 4) == round(math.dist(x,y), 4)
+    assert dist(x, y) == dist(y, x)
+    assert dist(x, x) == 0
 
 # TODO add a function that get data frame as an argument and return it after some preprocess/change
+def change_df(df: pd.DataFrame) -> pd.DataFrame:
+    df_copy = df.copy()
+    df_copy.iloc[:,2] = 5
+    return df_copy
+
+@pytest.fixture
+def rnd_data_frame(scope="module"):
```

Comment:
This is the API of scope module fixture
```suggestion
@pytest.fixture(scope="module")
def rnd_data_frame():
```